src/additional_functions/helper.jl

```diff
@@ -108,10 +108,7 @@ function sparse_outer_mul!(C, A, B::Vector, ind) #computes A*S*B -> C, where ind
 end
 
 function cov_and_mean(rows; corrected = false)
-    data = transpose(reduce(hcat, rows))
-    size(rows, 1) > 1 ? obs_cov = Statistics.cov(data; corrected = corrected) :
-    obs_cov = reshape([0.0], 1, 1)
-    obs_mean = vec(Statistics.mean(data, dims = 1))
+    obs_mean, obs_cov = StatsBase.mean_and_cov(reduce(hcat, rows), 1, corrected = corrected)
```

Comment:
I think it has to be this way around:
```suggestion
    obs_mean, obs_cov = StatsBase.mean_and_cov(reduce(hcat, rows), 2, corrected = corrected)
```